### PR TITLE
Optimize tensor.slice()

### PR DESCRIPTION
**Pull request by Honry**
The performance of executing `tensor.slice()` is super poor, especially for the 'logits' tensor with large dimensions.

```
const logits = outputs.logits.slice(null, -1, null);`
```

This is because currently implementation of the `slice` method manually iterates through each element and calculate indices which is a big time consuming if the tensor shape is large.

For cases like `slice(null, -1, null)`, where the slicing operation is contiguous along certain dimensions, which can be optimized by bulk copy by using `TypeArray.subarray()` and `TypeArray.set()`.

### CODE REVIEW

**Comment by Honry** [comment]
@xenova, PTAL, thanks!

On my test device, this can improve the execution time for `const logits = outputs.logits.slice(null, -1, null);` from 11ms to 0.3ms for Qwen model (with logits shape: `[batch_size,sequence_length,151936]`).

cc/ @ibelem

**Comment by xenova** [comment]
Oh wow, this looks like a great PR! Running tests and reviewing now!

Thanks @Honry

**Comment by xenova** [comment]
Very impressive: ran this with qwen3 and got a pretty substantial TPS increase 🙌 

Before: 45.087279331416916 tokens per second
After: 50.945617704855884 tokens per second

+13% speed boost to generate 512 tokens 🔥 

Merging now!
